src/Bicep.Core.IntegrationTests/ParameterFileTests.cs

```diff
@@ -234,4 +244,45 @@ invalid file
             ("BCP104", DiagnosticLevel.Error, "The referenced module has errors."),
         });
     }
+
+    [TestMethod]
+    public void ExternalInput_assigned_to_parameter()
+    {
+        var result = CompilationHelper.CompileParams(
+            ServicesWithExternalInputFunctionEnabled,
+("parameters.bicepparam", @"
+using none
+param foo = externalInput('sys.cli', 'foo')
+"));
+
+        result.Should().NotHaveAnyDiagnostics();
+        var parameters = TemplateHelper.ConvertAndAssertParameters(result.Parameters);
+        parameters["foo"].Value.Should().BeNull();
+        parameters["foo"].Expression.Should().DeepEqual("""[externalInputs('0')]""");
+
+        var externalInputs = TemplateHelper.ConvertAndAssertExternalInputs(result.Parameters);
+        externalInputs["0"].Should().DeepEqual(new JObject
+        {
+            ["kind"] = "sys.cli",
+            ["options"] = "foo",
+        });
+    }
+
+    [TestMethod]
+    public void ExternalInput_non_compile_time_constant_is_blocked()
+    {
+        var result = CompilationHelper.CompileParams(
+            ServicesWithExternalInputFunctionEnabled,
+("parameters.bicepparam", @"
+using none
+var myVar = 2 + 3
+param foo = externalInput('sys.cli', myVar)
```

Comment:
I'm fine blocking this in the initial PR, but shouldn't we be able to support cases like this where we're able to evaluate the expression?

There are of course cases which would have to remain blocked - e.g.:
```bicep
var foo = externalInput('sys.cli', 'foo')
param bar = externalInput('sys.cli', foo)
```

Reply:
addressed offline. Ideally, we should be able to support cases like the one mentioned above, but currently it is complicated to make that happen. Will revisit later.